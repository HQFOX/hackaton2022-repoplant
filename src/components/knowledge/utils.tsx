import {
  AssetInventoryMetadata,
  AssetInventoryConfiguration,
} from "@hitachivantara/uikit-react-core";

const metadata: AssetInventoryMetadata[] = [
  {
    id: "id1",
    accessor: "title",
    cellType: "alpha-numeric",
  },
  {
    id: "id2",
    accessor: "project",
    cellType: "alpha-numeric",
  },
];

const viewConfiguration: AssetInventoryConfiguration = {
  breakpoints: { xs: 12, sm: 6, md: 4, lg: 3, xl: 3 },
  columnConfiguration: [
    {
      style: { paddingLeft: "10px", textAlign: "start" },
    },
    {
      title: "Title",
      style: { textAlign: "start", minWidth: "120px" },
    },
    {
      title: "Project",
      style: { textAlign: "start", minWidth: "80px" },
    },
    {
      title: "Information Type",
      style: { textAlign: "start", minWidth: "60px" },
    },
    {
      title: "Uploaded By",
      style: { textAlign: "start", minWidth: "60px" },
    },
    {
      title: "Date",
      style: {
        textAlign: "start",
      },
    },
  ],
};

const configuration = {
  metadata,
  viewConfiguration,
};

const actions = [{ id: "view", label: "View", disabled: false }];

export { configuration, actions };
