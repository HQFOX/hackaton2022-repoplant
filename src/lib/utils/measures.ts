export const avg = (values: number[]): number => {
  if (!values || values.length === 0) {
    return 0;
  }
  return (
    values.reduce((a, b) => {
      return a + b;
    }) / values.length
  );
};
