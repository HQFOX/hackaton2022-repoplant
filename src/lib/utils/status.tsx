import React, { lazy } from "react";
import { Level5 } from "@hitachivantara/uikit-react-icons";
import { Semantic } from "@hitachivantara/uikit-react-lab";

export const getStatus = (status: any) => {
  const Level = `Level${status}`;
  const statusObj: {
    Icon: React.ReactNode;
    sema: Semantic;
  } = { Icon: <Level5 />, sema: "sema5" };
  return statusObj;
};
