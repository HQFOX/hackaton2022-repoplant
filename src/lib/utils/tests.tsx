import React from "react";
import { MemoryRouter } from "react-router-dom";
import GlobalProvider from "lib/providers/GlobalProvider";

export const withProvider = <P extends {}>(
  Component: React.ComponentType<P>
): React.FC<P> => (props) => (
  <MemoryRouter initialEntries={[{ pathname: "/", key: "testKey" }]}>
    <GlobalProvider>
      <Component {...props} />
    </GlobalProvider>
  </MemoryRouter>
);
