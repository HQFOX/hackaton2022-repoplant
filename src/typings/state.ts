import { AuthState } from "./auth";

export type DeepPartial<T> = T extends Function
  ? T
  : T extends object
  ? { [P in keyof T]?: DeepPartial<T[P]> }
  : T;

// The top-level state object
export interface AppState {
  readonly auth: AuthState;
}

export type TestAppState = DeepPartial<AppState>;
