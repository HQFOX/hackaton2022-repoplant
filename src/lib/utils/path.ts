import { Page } from "typings/pages";

const getSelection = (data: Page[], pathname: string): Page | undefined => {
  let selection;

  data.forEach((item) => {
    if (item.path === pathname) selection = { ...item };
    else if (item.data && pathname.indexOf(item.path) > -1)
      selection = getSelection(item.data, pathname);
  });

  return selection;
};

export { getSelection };
