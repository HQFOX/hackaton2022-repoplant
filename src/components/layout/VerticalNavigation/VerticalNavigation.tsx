import React, { useContext } from "react";
import { useLocation } from "react-router-dom";
import { useMediaQuery, useTheme } from "@material-ui/core";
import { LogOut, User } from "@hv/uikit-react-icons";
import {
  HvVerticalNavigation,
  HvVerticalNavigationTree,
  HvVerticalNavigationActions,
  HvVerticalNavigationAction,
} from "@hv/uikit-react-core";
import history from "lib/utils/history";
import NavigationContext from "lib/NavigationContext";
import { pages } from "lib/api/pages";
import { getSelectedPage } from "lib/utils/path";

const VerticalNavigation = ({ logout }) => {
  const theme = useTheme();
  const isMdUp = useMediaQuery(theme.breakpoints.up("md"));
  const { isOpen, toggleOpen } = useContext(NavigationContext);

  const { pathname } = useLocation();
  const selectedPage = getSelectedPage(pages, pathname);

  const handleChange = (event, selection) => {
    if (selection.path) history.push(selection.path);
  };

  if (!isOpen || isMdUp) return null;

  return (
    <HvVerticalNavigation
      isCollapsable={false}
      isOpen={isOpen}
      toggleOpenCallback={toggleOpen}
    >
      <HvVerticalNavigationTree
        data={pages}
        selected={selectedPage?.id}
        onClick={handleChange}
      />
      {!isMdUp && (
        <HvVerticalNavigationActions>
          <HvVerticalNavigationAction
            label="Profile"
            icon={<User />}
            onClick={() => {}}
          />
          <HvVerticalNavigationAction
            label="Logout"
            icon={<LogOut />}
            onClick={() => logout()}
          />
        </HvVerticalNavigationActions>
      )}
    </HvVerticalNavigation>
  );
};

export default VerticalNavigation;
