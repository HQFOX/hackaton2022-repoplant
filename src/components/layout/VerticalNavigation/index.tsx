import { withStyles } from "@material-ui/core";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { logout } from "store/auth/thunks";
import styles from "./styles";
import VerticalNavigation from "./VerticalNavigation";

const mapStateToProps = ({ auth }) => ({
  auth,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      logout,
    },
    dispatch
  );

export default withStyles(styles, { name: "VerticalNavigation" })(
  connect(mapStateToProps, mapDispatchToProps)(VerticalNavigation)
);
