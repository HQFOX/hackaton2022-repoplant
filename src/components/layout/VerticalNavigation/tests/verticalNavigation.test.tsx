import React from "react";
import { mount } from "enzyme";
import { withStoreProvider } from "lib/utils/tests";
import { TestAppState } from "typings/state";
import VerticalNavigation from "..";

const pageData = [
  {
    id: "1",
    label: "Overview",
    path: "/overview"
  },
  {
    id: "2",
    label: "Templates",
    path: "/templates"
  }
];

describe("<VerticalNavigation />", () => {
  let component;

  const WithProvider = withStoreProvider(VerticalNavigation, {
    router: { location: { pathname: "/" } },
    auth: { isAuthed: false },
    pages: { data: pageData },
    logout: jest.fn()
  } as TestAppState);

  beforeEach(() => {
    component = mount(<WithProvider />);
  });

  it("should be defined", () => {
    expect(component).toBeDefined();
  });

  it("matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("should be rendered", () => {
    const navigation = component.find(VerticalNavigation);
    expect(navigation.length).toBe(1);
  });
});
