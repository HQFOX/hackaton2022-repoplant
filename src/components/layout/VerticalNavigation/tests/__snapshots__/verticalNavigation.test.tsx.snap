// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalNavigation /> renders the component 1`] = `
<body
  class="js-focus-within"
>
  <div />
</body>
`;
